### ISSUE TITLE
BlitScaled on window surface causes unintialized values to be used

### ISSUE THREAD

**Issue by mr-sihc** [libsdl-org/SDL#8350]
when trying to draw into surface of window through commands like
`SDL_BlitScaled` (`SDL_BlitSurface` seems to be fine)
it will cause a LOT of errors, using uninitialized values.

I tried to find a very minimal test case and i found these things:
1. BlitScaled on it's own seems to be OK
2. Creating the window and then it seems, the first BlitScaled will be problematic

So with this program, when run in valgrind, will produce SO many errors, valgrind gives up on tracking them:
```c
#include <stdio.h>
#include <SDL2/SDL.h>

#include <assert.h>
#include <stdint.h>

uint8_t splash[]  = {
	2, 123, 234, 232
};

SDL_Window*  window;
SDL_Surface* window_surface;
SDL_Surface* splash_surface;

void SPLASH_Show() {
	window = SDL_CreateWindow(
		"Splash",
		SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
		1920, 1080,
		0
	);
	
	assert(window);
	
	//// copy splash into window surface
	// load splash from mem into surface
	splash_surface = SDL_CreateRGBSurfaceWithFormatFrom(
		splash,
		2, 2, sizeof(*splash), 2,
		SDL_PIXELFORMAT_RGB332
	);
	
	assert(splash_surface);
	
	window_surface = SDL_GetWindowSurface(window);
	
	// and copy from splash surface into window surface
	SDL_BlitScaled(splash_surface, NULL, window_surface, NULL);
	
	// and finish by updating surface
	SDL_UpdateWindowSurface(window);
}

void SPLASH_Hide() {
	SDL_FreeSurface(splash_surface);
	SDL_DestroyWindow(window);
	
	splash_surface = NULL;
	window = NULL;
}

int main() {
	SDL_Init(SDL_INIT_VIDEO);
	
	assert(SDL_WasInit(SDL_INIT_VIDEO));
	
	for (int i = 0; i < 2; i++) {
		SPLASH_Show();
		SPLASH_Hide();
	}
	
	SDL_Quit();
	
	return 0;
}
```

valgrind:
```
==26288== Use of uninitialised value of size 8
==26288==    at 0x4952FB6: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x492FC91: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x495F7F3: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x1092C6: SPLASH_Show (in /home/kaktus/testcrash.out)
==26288==    by 0x109385: main (in /home/kaktus/testcrash.out)
==26288== 
==26288== Use of uninitialised value of size 8
==26288==    at 0x4952FDB: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x492FC91: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x495F7F3: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x1092C6: SPLASH_Show (in /home/kaktus/testcrash.out)
==26288==    by 0x109385: main (in /home/kaktus/testcrash.out)
==26288== 
==26288== Use of uninitialised value of size 8
==26288==    at 0x4952FF1: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x492FC91: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x495F7F3: ??? (in /usr/lib/libSDL2-2.0.so.0.2800.3)
==26288==    by 0x1092C6: SPLASH_Show (in /home/kaktus/testcrash.out)
==26288==    by 0x109385: main (in /home/kaktus/testcrash.out)
==26288== 
==26288== 
==26288== More than 10000000 total errors detected.  I'm not reporting any more.
==26288== Final error counts will be inaccurate.  Go fix your program
```

this was run under ARTIX linux on X11 i3 (wayland tested too, under weston)

if anything more is needed, let me know.

**Comment by madebr** [comment]
This has been fixed in 4cd0c13823551cce279c9ff3d284c41c28deb3d8 (SDL3) and 2d5c05a69cca65e0eb01b4bae23737fe970f7bb1 (SDL2).

**Comment by mr-sihc** [comment]
thanks!
